src/vs/workbench/common/codecs/chatbotPromptCodec/tokens/fileReference.ts

```diff
@@ -0,0 +1,94 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { BaseToken } from '../../baseToken.js';
+import { Word } from '../../simpleCodec/tokens/index.js';
+import { assert } from '../../../../../base/common/assert.js';
+import { Range } from '../../../../../editor/common/core/range.js';
+
+// Start sequence for a file reference token in a prompt.
+const TOKEN_START: string = '#file:';
+
+/**
+ * A file reference token inside a prompt.
+ */
+export class FileReference extends BaseToken {
```

Comment:
I'm not sure `workbench/common` is the right place for these files, it all seems specific to the chat feature, so should it stay under `workbench/contrib/chat/...`?

Reply:
Moved 👍